### PR TITLE
Don't make humanize optional

### PR DESCRIPTION
**Pull request by ericdill**
This is a pretty annoying thing to have be optional, since it is part
of __repr__

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -6,5 +6,5 @@
       py_modules=['doct'],
       description='A read-only dottable dictionary',
       url='http://github.com/NSLS-II/doct',
-      requires=['six']
+      requires=['six', 'humanize']
```
Comment (tacaswell) [inline setup.py]: This is not the correct key.  One of our original `setup.py` files had this mistake and we have propagated it everywhere.  Should be `install_requires`

**Comment by tacaswell** [comment]
what is the license on humanize and how bad would it be to vendor the one function we use here?

**Comment by ericdill** [comment]
license is MIT. Does that mean we can vendor?

File: setup.py (hunk 1 above)
Comment (ericdill) [inline setup.py]: ugh yeah you're right. good catch!

**Comment by ericdill** [comment]
It is *very* ugly to vendor this one function.  It picks up at least 7 other functions and some locale stuff.  Very much :-1: on vendoring

**Comment by ericdill** [comment]
looks roughly like this: https://gist.github.com/ericdill/01f1824d012fe70cedb8bb9c203a9a77

**Comment by tacaswell** [comment]
Yeah, that vendoring might be bigger than this whole project.  Thanks for looking into that.

**Comment by ericdill** [comment]
Hah yeah its definitely a ~50% LoC increase

**Comment by tacaswell** [comment]
I have mixed feelings about this, but will not complain if someone else merges.

@klauer @danielballan

**Comment by ericdill** [comment]
Could you elaborate on why you don't like this?  We are not including `humanize` in *any* of our meta.yaml's, so any user of a jupyter notebook will hit this error if they try to look at the repr of a document given the following installation of a jupyterhub kernel:

```
conda install -c nsls2-tag analysis <beamline_configuration>
```

**Comment by tacaswell** [comment]
ah, only the normal `__str__` was guarded against humanize not being there, not the html.

I like the very minimal dependencies, but am failing to write a defend able justification other than 'just cuz' .

**Comment by tacaswell** [comment]
can you pull out the `try...except` in `__str__` and I will merge this.

**Comment by ericdill** [comment]
Would you rather we guard against the html repr having humanize? Future Eric wants to never see this error again.  I have no strong preferences regarding which path we go down (guard in html repr or make humanize a dependency)

**Comment by ericdill** [comment]
> can you pull out the try...except in __str__ and I will merge this.

done.

**Comment by tacaswell** [comment]
`vstr` apparently also depends on prettytable.

Glad I waited for travis there...

**Comment by ericdill** [comment]
Ah good catch...  Ok prettytable is now required too.

**Comment by ericdill** [comment]
Thanks!

**Comment by tacaswell** [comment]
now you _really_ have to build prettytable ;)

**Comment by ericdill** [comment]
How should I version bump for changing dependencies?  major/minor/micro?

**Comment by tacaswell** [comment]
Micro

**Pull request merged**